src/collection.ts

```diff
@@ -975,12 +998,45 @@ export class Collection<TSchema extends Document = Document> {
    */
   async count(filter: Filter<TSchema> = {}, options: CountOptions = {}): Promise<number> {
     return executeOperation(
-      this.s.db.s.client,
-      new CountOperation(
-        MongoDBNamespace.fromString(this.namespace),
-        filter,
-        resolveOptions(this, options)
-      )
+      this.client,
+      new CountOperation(this.fullNamespace, filter, resolveOptions(this, options))
     );
   }
+
+  listSearchIndexes(options?: ListSearchIndexesOptions): ListSearchIndexesCursor;
+  listSearchIndexes(indexName: string, options?: ListSearchIndexesOptions): ListSearchIndexesCursor;
+  listSearchIndexes(
+    indexNameOrOptions?: string | ListSearchIndexesOptions,
+    options?: ListSearchIndexesOptions
+  ): ListSearchIndexesCursor {
+    options =
+      typeof indexNameOrOptions === 'object' ? indexNameOrOptions : options == null ? {} : options;
+    const indexName =
+      indexNameOrOptions == null
+        ? null
+        : typeof indexNameOrOptions === 'object'
+        ? null
+        : indexNameOrOptions;
+
+    return ListSearchIndexesCursor.create(this, indexName, options);
+  }
+
+  async createSearchIndex(description: SearchIndexDescription): Promise<string> {
+    const indexes = await this.createSearchIndexes([description]);
+    return indexes[0];
+  }
+
+  async createSearchIndexes(
+    descriptions: ReadonlyArray<SearchIndexDescription>
+  ): Promise<string[]> {
+    return executeOperation(this.client, new CreateSearchIndexesOperation(this, descriptions));
+  }
+
+  async dropSearchIndex(name: string): Promise<void> {
+    return executeOperation(this.client, new DropSearchIndexOperation(this, name));
+  }
+
+  async updateSearchIndex(name: string, definition: Document): Promise<void> {
```

Comment:
Sorry to jump in here, but wasn't there also going to be a `get` operation to just get one by name?

Reply:
I saw someone mention it in passing in slack _somewhere_, but now can't find it and I'm kinda wondering if I'm imagining it. If it isn't listed somewhere, then it probably isn't a thing.

Reply:
Are you thinking of the optional overload for `create`?  I chose not to implement this in Node to align with the existing `createIndex` API in the driver

```
createSearchIndex(name: String, definition: Document)
```

Reply:
I checked again and I'm now pretty sure I was imagining it.